src/include/kdbmacros.h

```diff
@@ -37,24 +37,33 @@
 	do                                                                                                                                 \
 	{                                                                                                                                  \
 		if (errno == EACCES)                                                                                                       \
-			ELEKTRA_SET_ERROR (109, parentKey, strerror (errno));                                                              \
+			ELEKTRA_SET_RESOURCE_ERRORF (                                                                                      \
+				parentKey,                                                                                                 \
+				"Insufficient permissions to open configuration file for reading. Reason: %s. You might want "             \
+				"to retry as root or change access using chmod.",                                                          \
+				strerror (errno));                                                                                         \
 		else                                                                                                                       \
-			ELEKTRA_SET_ERROR (110, parentKey, strerror (errno));                                                              \
+			ELEKTRA_SET_RESOURCE_ERRORF (parentKey, "Could not open configuration file for reading. Reason: %s",               \
```

Comment:
Which file?

Reply:
We agreed that eventual error message improvements will be done in another PR. This PR here has already again lots of comments

Reply:
I thought that this PR is already the other PR. If you set "ready to merge" I expect **ALL** the code in the PR to be ready to merge. Please do not submit such large PRs for review if you do not want them to be reviewed, this wastes a lot of time on both sides.

Reply:
We agreed in the elektra meeting to keep PR's as little as possible. If I also fix *every of the 650* messages this will PR will get again out of hand ... We *definitely* agreed that this PR does only affect the error categorizations, not the message in addition

Reply:
The PR is not getting bigger if the messages are better. But ok, if you prefer to make it in a different PR, ok.

Reply:
I **highly** doubt this because up until now each and every PR which I had done with even minor subjective matters in it like wording, adding extra information, removing adding "sorry", etc gained tons of comments. I will not do that again.

Reply:
> removing adding "sorry"

This is not a good example as on this topic there are papers.

> I will not do that again.

I would very much welcome if the quality of your PRs would improve even more.

Reply:
I changed the text to include the filename. It now gives a message as following: ```Resource: Insufficient permissions to open configuration file <path> for reading. Reason: Permission denied. You might want to retry as root or change access using chmod.```

Reply:
Thank you! Btw. the solution you do not need to add in the text as it will be added differently later. (Otherwise we mind and up with many duplicates of the text "You might want to retry...".)

Only make sure that everything visible in the PR has very good quality.